### PR TITLE
Huffman_encoding&decoding

### PR DESCRIPTION
**Pull request by barnamensch056**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes
### Description

(Write your answer here.)
Huffman coding of a string by encoding and decoding using tree and heap.
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: C/cp/BoyerMooreVote.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/*C code to find the majority element in a given set of string by Boyer-Moore voting algorithm */
```
Comment (atarax665) [inline C/cp/BoyerMooreVote.c]: Remove this file and pull a separate request from a new branch regarding this program.

File: C/cp/Longest_substring_without_repeating_chars.cpp [hunk 2]
```diff
@@ -61,16 +61,11 @@ int main(void)
 
 
 /* Sample Input-output
-
```
Comment (atarax665) [inline C/cp/Longest_substring_without_repeating_chars.cpp:64]: Revert these changes as I don't think they are necessary.

File: Python/other/Huffman_coding.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+'''Python code to implement HUffman encoding
```
Comment (atarax665) [inline Python/other/Huffman_coding.py]: This is Huffman encoding algorithm only, edit the docstring. Codes contributed in python are expected to follow our [Python code template](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/Code_Template.md) to ease the reviewing process and maintain the quality of contributed code.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/other/Huffman_coding.py (hunk 3 above)
Comment (barnamensch056) [reply Python/other/Huffman_coding.py]: Surely I will look into the matter and make the changes.

**Comment by barnamensch056** [comment]
@atarax665 is that ok now?

**Comment by HarshCasper** [comment]
@barnamensch056 Make README changes as well. Once done, push an atomic commit instead of multiple commits.

**Comment by barnamensch056** [comment]
> @barnamensch056 Make README changes as well. Once done, push an atomic commit instead of multiple commits.
Yeah @HarshCasper I have made all the necessary changes(readme also) and you can merge the PR now.

**Comment by barnamensch056** [comment]
> @barnamensch056 Make README changes as well. Once done, push an atomic commit instead of multiple commits.
 Actually I have committed  the changes in github and don't know how to access them through git, tell me if needed.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Do not delete ```Longest_substring_without_repeating_chars.cpp```, just revert the changes that you committed to that file and please update README.md.

**Comment by barnamensch056** [comment]
Sorry for the inconvinience in updating the readme now deepsource found conflict :'/

**Comment by HarshCasper** [comment]
@barnamensch056 Please resolve the Merge Conflicts. You can safely ignore DeepSource issues for now.

**Comment by barnamensch056** [comment]
Sorry @HarshCasper Sir but the branch cannot be merged by github can you tell how to do this by git, actually so far after opening the PR I have committed changes through github :'-/

**Comment by HarshCasper** [comment]
> Sorry @HarshCasper Sir but the branch cannot be merged by github can you tell how to do this by git, actually so far after opening the PR I have committed changes through github :'-/

The Command Line Instructions are given below.

**Comment by HarshCasper** [comment]
Are you working on it @barnamensch056 ?

**Comment by barnamensch056** [comment]
Yes but I don't know the solution @_@

**Comment by HarshCasper** [comment]
I would close it then. Feel free to make another PR

**Comment by barnamensch056** [comment]
Sure I'll make another clean PR with the Huffman coding python file and update the readme too :) thanks @HarshCasper and @atarax665 for this journey.
